### PR TITLE
3116 fixes Firefox form POST issues

### PR DESCRIPTION
**Pull request by payten**
Delivers #3116.

This fixes a Firefox bug, whereby navigating the GradebookNG tool and returning to the Grades sepreadsheet causes all subsequent POST requests to fail with a 302 redirect.

To fix, set mount points for the 4 GradebookNG pages, so the base page is always consistent.  Secondly, use the class variant of setResponsePage to navigate to those mount points.

I haven't set a milestone yet.. 11.2 so we can get some reasonable testing in?

### CODE REVIEW

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/PermissionsPage.java [hunk 1]
```diff
@@ -58,18 +59,21 @@ public PermissionsPage() {
...
-		this.taSelected = taSelected;
-	}
-
 	@Override
 	public void onInitialize() {
 		super.onInitialize();
 
+		// grab the selected parameter (if provided)
+		final PageParameters params = getPageParameters();
+		final Optional<String> taUuid = Optional.ofNullable(params.get("selected").toOptionalString());
```
Comment (steveswinsburg) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/PermissionsPage.java]: is the Optional required? Can't this just be a String?

Comment (payten) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/PermissionsPage.java]: Yeah, it doesn't need to be an Optional, I only made it one to match up with `taSelected` which is now an Optional... it also felt like a candidate, as it helps ensure we avoid a NPE.  Happy to refactor?

Comment (steveswinsburg) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/PermissionsPage.java]: If its clearer that its just a string that might have a value or not then maybe this should just be a string, likewise `taSelected`. Might be a bit overkill to wrap it all in an Optional when we can just `StringUtils.IsBlank` it?

Comment (payten) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/PermissionsPage.java]: Sure thing, I'll give it a tinker and poke you when it's up.

**Comment by payten** [comment]
Hi @steveswinsburg, I've removed the use of `Optional` and use StringUtils.isBlank when appropriate plus returned the original null checks.

**Comment by steveswinsburg** [comment]
Nice one ⭐

**Comment by steveswinsburg** [comment]
Added this to 11.2 so we dont forget about it. We can push to a later version if required 👍

**Comment by payten** [comment]
Great, thanks 🍻

**Pull request merged**